digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/ComplementaryLinkFilterDAO.scala

```diff
@@ -105,78 +106,88 @@ class ComplementaryLinkDAO {
 
       val linkSource = LinkGeomSource.ComplementaryLinkInterface
 
-      RoadLinkFetched(linkId, municipalityCode, geometry, administrativeClass,
-         trafficDirection, featureClass , modifiedAt, attributes,
-        lifecycleStatus, linkSource, length)
+//      RoadLink(linkId, municipalityCode, geometry, administrativeClass,
+//         trafficDirection, featureClass , modifiedAt, attributes,
+//        lifecycleStatus, linkSource, length)
+//
+      RoadLink(linkId, geometry, length, administrativeClass,
+        -1 // FunctionalClass pois
+        , TrafficDirection.UnknownDirection, //TrafficDirection pois?
+        UnknownLinkType, // RoadLink pois
+        modifiedAt,
+        None
+        , attributes,
+        lifecycleStatus, linkSource)
+
     }
   }
 
-  def fetchByLinkId(linkId: String): Option[RoadLinkFetched] = {
+  def fetchByLinkId(linkId: String): Option[RoadLink] = {
     fetchByLinkIds(Set(linkId)).headOption
   }
 
-  def fetchByLinkIds(linkIds: Set[String]): List[RoadLinkFetched] = {
+  def fetchByLinkIds(linkIds: Set[String]): List[RoadLink] = {
     time(logger, "Fetch complementary data by linkIds") {
       val sql = s"""SELECT * FROM complementary_link_table WHERE id IN (${linkIds.map(lid => "'" + lid + "'").mkString(", ")})"""
-      withDynTransaction(Q.queryNA[RoadLinkFetched](sql).list)
+      withDynTransaction(Q.queryNA[RoadLink](sql).list)
     }
   }
 
-  def fetchByLinkIdsF(linkIds: Set[String]): Future[Seq[RoadLinkFetched]] = {
+  def fetchByLinkIdsF(linkIds: Set[String]): Future[Seq[RoadLink]] = {
     Future(fetchByLinkIds(linkIds))
   }
 
   /**
      * Returns RoadLinks by municipality.
      */
-  def queryByMunicipality(municipality: Int, filter: Option[String] = None): Seq[RoadLinkFetched] = {
-    val filterString = filter.getOrElse("")
+  def queryByMunicipality(municipality: Int, filter: Option[String] = None): Seq[RoadLink] = {
+    val filterString = if (filter.isDefined) " AND" + filter.get.replaceFirst("(?i)AND", "") else ""
```

Comment:
tää näyttää hassulta, mitä tässä tapahtuu? 

Mistä tuo filtteri tulee? 
Ja miksi tässä poistetaan filtteristä AND ja lisätään siihen eteen AND?

Reply:
Lähinnä juuri tuo kun ei ollut varmuutta mistä tulee ja jos siinä on and niin ei tule dublikaattina.

Reply:
Mielummin selvitetään mistä filtteri tulee ja korjataan siellä virheellisesti syötetty sql koodinpätkä

Reply:
Muutenhan tähän voisi laittaa

`if (filter.isDefined)  filter.get  else ""`